clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -40,15 +48,29 @@ public KafkaMetric(Object lock, MetricName metricName, MetricValueProvider<?> va
         this.time = time;
     }
 
+    /**
+     * Get the configuration of this metric.
+     * This is supposed to be used by server only.
```

Comment:
What do you mean by `This is supposed to be used by server only.`?
Same for the other method below

Reply:
@mimaison it seems to me those methods are used by server to "write" something (for example, quota), and so users should view the metrics as readonly object. Please correct me If I misunderstand the purpose.